lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -69,10 +69,41 @@ struct BroadcastOpConversion
     auto srcOffsets = emitOffsetForLayout(srcLayout, srcShape);
     auto resultOffsets = emitOffsetForLayout(resultLayout, resultShape);
     SmallVector<Value> srcVals = getElementsFromStruct(loc, src, rewriter);
+    if (auto srcMma = srcLayout.dyn_cast<MmaEncodingAttr>()) {
+
+      // NOTE: This is just an naive fix, but for MMA layout, and 2-d fix should
+      // be all right.
+      // TODO[Superjomn]: Replace this with a generic implementation.
+      if (srcMma.isVolta() &&
+          (srcTy.getElementType().isF16() || srcTy.getElementType().isBF16())) {
```

Comment:
nit: I don't think the logic here really depends on element type to be F16/BF16. And perhaps assert is better here, like:
if (srcMma.isVolta()) {
   assert(isF16 || isBF16);
    ...
}